net/src/memory_server.rs

```diff
@@ -555,34 +761,103 @@ impl InMemoryServer {
     /// this in-memory module routes it to the first node connected on that dna.
     /// this works because we also send store requests to all connected nodes.
     fn priv_serve_FetchMeta(&mut self, msg: &FetchMetaData) -> NetResult<()> {
+        // Requester must be tracking
+        let sender_info = Some((msg.requester_agent_id.clone(), Some(msg.request_id.clone())));
+        let is_tracking =
+            self.priv_check_or_fail(&msg.dna_address, &msg.requester_agent_id, sender_info)?;
+        if !is_tracking {
+            return Ok(());
+        }
         match self.senders_by_dna.entry(msg.dna_address.to_owned()) {
+            // Relay fetchMeta to first agent tracking this DNA
             Entry::Occupied(mut e) => {
                 if !e.get().is_empty() {
-                    let r = &e.get_mut()[0];
+                    let (_k, r) = &e
+                        .get_mut()
+                        .iter()
+                        .next()
+                        .expect("senders_by_dna.entry does not hold any value");
                     r.send(JsonProtocol::HandleFetchMeta(msg.clone()).into())?;
                     return Ok(());
                 }
             }
+            // TODO: Once linking meta is properly implemented in Core
```

Comment:
What exactly does this mean?
Links (and the according meta messages and handling thereof) are implemented already.

Reply:
If you look into `publish_link_meta()` you'll see that Core is storing the whole AddLinkEntry with its meta as the value of the link and the attribute of this meta is always "link".  I'm pretty sure thats not how we want links to work. We would want the attribute to be the type of the link (as defined by the zome code) and the value to be the hash of the other Entry.

Reply:
No, we can't do it the way you say because that would lock-in (or equate) the meta system with links. But we have more meta items, like CRUD status and links, and potentially more to come. We need to send the whole `AddLinkEntry` as meta value since the receiving node needs that info, esp. the header of that entry, to run the validation on that StoreMeta request.

So, the way it is implemented right now is pretty much the way it should be, I think.

Reply:
The reason we need `PublishMeta` as a network method at all is that the routing (to the nodes that receive the `HandleStoreMeta`) needs to look at the address of a separately defined `base` and not the thing that gets send. - It was not meant as a 1-to-1 match of the EAVs or even links.

Reply:
I think we should talk about this. The meta system is already locked in for links since linking is the main feature of Holochain's DHT. In the current way, the network module and thus DHT cannot optimize requests on link type/tag since this is hidden within the meta value. For example, when requesting a link of a certain type/tag, the DHT will have to fetch ALL links on that entry from the network and let Core do the triage.
Also some other parts of the code seem to be expecting what I'm saying: Look at `reduce_get_links_inner()`. Its asking for meta with the attribute `link__{}`. This attribute name is never used when doing a PublishMeta, so I'm pretty sure the 'get_links' isn't getting anything back or if it is, its not based on that attribute.

Reply:
What do you mean with „it is already locked in for links“? I can’t see how this could be the case as we are also storing (and publishing) CRUD statuses as meta items. We recently discussed that we might have to store headers as well as meta items on links. The last point I see (there was no explicit discussion nor decision around that yet, but I can’t imagine how we could get around this) is validation packages that would need to be stored as meta.

Reply:
I‘m pretty sure that links get published with ˋlink__<tag>ˋ.

Reply:
Yeah, I think we should sync ASAP. When you talk about optimization, you seem to be asserting things that go beyond the design @neonphog and I discussed.

I expect the network implementation to tunnel through the meta attribute and content to the „right“ other instance, where core understands the attribute and deals with it accordingly. Same as with the entries in a regular publish. The only difference (from the network implementation) between Publish and PublishMeta is that the latter routes according to a base address, different to the content that is published. And it has an attribute that gets passed through as, so that the receiving core instances can understand different meta items like links, CRUD and the others mentioned above. @ddd-mtl